lib/internal/fs/promises.js

```diff
@@ -678,6 +695,18 @@ async function writeFile(path, data, options) {
   return PromisePrototypeFinally(writeFileHandle(fd, data), fd.close);
 }
 
+function isCustomIterable(obj) {
+  if (obj == null) {
```

Comment:
`in` throws when used on `undefined`, `null`, a string or a number. It'd be nice to add test cases for those values.

```suggestion
  if (typeof !== 'object' || obj === null) {
```